packages/react-dom-bindings/src/server/ReactDOMServerFormatConfig.js

```diff
@@ -2001,65 +2034,172 @@ export function writeEndSegment(
+//     }
+//     // Find the boundary around the fallback. This is always the previous node.
+//     const suspenseNode = suspenseIdNode.previousSibling;
+
+//     if (contentNode) {
+//       // Clear all the existing children. This is complicated because
+//       // there can be embedded Suspense boundaries in the fallback.
+//       // This is similar to clearSuspenseBoundary in ReactDOMHostConfig.
+//       // TODO: We could avoid this if we never emitted suspense boundaries in fallback trees.
+//       // They never hydrate anyway. However, currently we support incrementally loading the fallback.
+//       const parentInstance = suspenseNode.parentNode;
+//       let node = suspenseNode.nextSibling;
+//       let depth = 0;
+//       do {
+//         if (node && node.nodeType === COMMENT_NODE) {
+//           const data = node.data;
+//           if (data === SUSPENSE_END_DATA) {
+//             if (depth === 0) {
+//               break;
+//             } else {
+//               depth--;
+//             }
+//           } else if (
+//             data === SUSPENSE_START_DATA ||
+//             data === SUSPENSE_PENDING_START_DATA ||
+//             data === SUSPENSE_FALLBACK_START_DATA
+//           ) {
+//             depth++;
+//           }
+//         }
 //
-//   const endOfBoundary = node;
+//         const nextNode = node.nextSibling;
+//         parentInstance.removeChild(node);
+//         node = nextNode;
+//       } while (node);
 //
-//   // Insert all the children from the contentNode between the start and end of suspense boundary.
-//   while (contentNode.firstChild) {
-//     parentInstance.insertBefore(contentNode.firstChild, endOfBoundary);
+//       const endOfBoundary = node;
+//
+//       // Insert all the children from the contentNode between the start and end of suspense boundary.
+//       while (contentNode.firstChild) {
+//         parentInstance.insertBefore(contentNode.firstChild, endOfBoundary);
+//       }
+//
+//       suspenseNode.data = '$';
+//     } else {
+//       suspenseNode.data = '$!';
+//       suspenseIdNode.setAttribute('data-dgst', errorDigest)
+//     }
+//
+//     if (suspenseNode._reactRetry) {
+//       suspenseNode._reactRetry();
+//     }
 //   }
-
-//   suspenseNode.data = SUSPENSE_START_DATA;
-//   if (suspenseNode._reactRetry) {
-//     suspenseNode._reactRetry();
+//   return function completeBoundary(suspenseBoundaryID, contentID, styleResources){
+//     const contentNode = document.getElementById(contentID);
+//     // We'll detach the content node so that regardless of what happens next we don't leave in the tree.
+//     // This might also help by not causing recalcing each time we move a child from here to the target.
+//     contentNode.parentNode.removeChild(contentNode);
+//     if (styleResources) {
+//       const p = $RR(styleResources);
+//       if (p) {
+//         return Promise.all(p).then(
+//           flipBoundary.bind(null, suspenseBoundaryID, contentNode, null),
```

Comment:
Instead of making an inner function for flipBoundary you can just self-reference this function and call it again with the `suspenseBoundaryID` and this time without the `styleResources`.